rql.go

```diff
@@ -453,7 +466,7 @@ func (p *parseState) field(f *field, v interface{}) {
 	if !ok {
 		must(f.ValidateFn(v), "invalid datatype for field %q", f.Name)
```

Comment:
ValueFn must be used here to support different data types between filter and underlying model, i.e:
`must(f.ValidateFn(f.ValueFn(v)), "invalid datatype for field %q", f.Name)`